addons/payment/models/payment_provider.py

```diff
@@ -315,7 +319,7 @@ def _check_fee_var_within_boundaries(self):
         :return None
         """
         for provider in self:
-            if any(not 0 <= fee < 100 for fee in (provider.fees_dom_var, provider.fees_int_var)):
+            if any(not 0 <= fee < 1 for fee in (provider.fees_dom_var, provider.fees_int_var)):
```

Comment:
Why isn't it a SQL constraint?

Reply:
(actually 2)

Reply:
It's fine as-is. Please r+ https://github.com/odoo/upgrade/pull/4106

Reply:
This is not a satisfying answer.

Reply:
I do believe that "It's fine as-is" is a satisfying answer to a question that I read as "Why don't you refactor this?" considering that it is, indeed, just fine as-is, but, to clarify:
- This task is about converting a float field to a monetary, which has nothing to do with dealing with SQL vs. Python constraints.
- The scope and range of this task were dimensioned to make it a good first task for a new dev. The goal is not to rewrite every unrelated method that shows up in the diff, in particular methods in which we change 2 characters.
- A Python constraint is easier to read and maintain than an SQL constraint.
- The performance gain of converting the constraint would be completely trivial here.
- This PR has been open for some time, and I'll like to be merged ASAP unless a *blocking* problem is found.

Reply:
Now, with these arguments, it's more precise.
It's OK not doing it in this PR but this is clearly the kind of constraint that should be done at the database level.